agent/bench-scripts/postprocess/process-iteration-samples

```diff
@@ -88,38 +88,34 @@ my $keep_failed_tool_data = $ARGV[6]; # (y/n) keep the tool data for a set of sa
 
 my $nr_significant_figures = 4;
 
+my $uid_label = get_label('uid_label');
+my $value_label = get_label('value_label');
+my $ts_label = get_label('timeseries_label');
+my $samples_label = get_label('samples_label');
+
 sub get_avg_stddev {
 	my $samples_start_index = shift;
-        my @numbers = @_;
+	my @numbers = @_;
 	my $count = scalar @numbers;
 	my $sum = 0;
 
 	if ($count == 1) {
-		return ($numbers[0]{get_label('value_label')}, 0, 0, $samples_start_index);
+		return ($numbers[0]{$value_label}, 0, 0, $samples_start_index);
 	} elsif ($count == 0) {
 		return (0, 0, 0, 0);
 	} else {
-		my $val;
 		my $sum;
-		my $index;
-		my $sqdiff;
+		foreach my $number (@numbers) {
+			$sum += $number->{$value_label};
+		}
+		my $avg = $sum / $count;
+
 		my $closest_index;
 		my $mindiff;
 		my $totalsqdiff;
-		my $stddev;
-		my $stddevpct;
-		my $avg;
-		my $i;
-		for ($i=0; $i < scalar @numbers; $i++) {
-			$val = $numbers[$i]{get_label('value_label')};
-			$sum += $val;
-		}
-		$avg = $sum / $count;
-
-		$index = $samples_start_index;
-		for ($i=0; $i < scalar @numbers; $i++) {
-			$val = $numbers[$i]{get_label('value_label')};
-			$sqdiff = ($avg - $val)**2;
+		my $index = $samples_start_index;
+		foreach my $number (@numbers) {
+			my $sqdiff = ($avg - $number->{$value_label})**2;
```

Comment:
We could combine the two loops into one if all we were interested in were the mean and the std deviation, but I don't see a way to get the `closest_index` with a single loop: we need to know the mean beforehand.

Reply:
Excellent!  I _hoped_ that there was a good reason why the loops were separate -- thanks for chasing that down, Nick!